src/test/java/io/xstefank/wildfly/bot/PRMentionsReviewsCombinationTest.java

```diff
@@ -0,0 +1,212 @@
+                    Mockito.verify(mocks.pullRequest(gitHubJson.id()), Mockito.never()).requestReviewers(ArgumentMatchers.anyList());
+                });
+    }
+
+    @Test
+    public void testMentionsReviewsReviewsHitSameRule() throws IOException {
+        wildflyConfigFile = """
+                wildfly:
+                  rules:
+                    - id: "test"
+                      title: "WFLY"
+                      directories: [src]
+                      notify: [user1, user2]
+                  format:
+                    title:
+                      enabled: false
+                    commit:
+                      enabled: false
+                """;
+        gitHubJson = GitHubJson.builder(TestConstants.VALID_PR_TEMPLATE_JSON)
+                .title(INVALID_TITLE)
+                .build();
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile("src/main/java/resource/application.properties")
+                .collaborators(Set.of("user1", "user2"));
+
+        given().github(mocks -> Util.mockRepo(mocks, wildflyConfigFile, gitHubJson, mockedContext))
+                .when().payloadFromString(gitHubJson.jsonString())
+                .event(GHEvent.PULL_REQUEST)
+                .then().github(mocks -> {
+                    Mockito.verify(mocks.pullRequest(gitHubJson.id()), Mockito.never())
+                            .comment(ArgumentMatchers.anyString());
+                    ArgumentCaptor<List<GHUser>> captor = ArgumentCaptor.forClass(List.class);
+                    Mockito.verify(mocks.pullRequest(gitHubJson.id())).requestReviewers(captor.capture());
+                    Assertions.assertEquals(captor.getValue().size(), 2);
+                    MatcherAssert.assertThat(captor.getValue().stream()
+                            .map(GHPerson::getLogin)
+                            .toList(), Matchers.containsInAnyOrder("user1", "user2"));
+                });
+    }
+
+    @Test
+    public void testMentionsReviewsReviewsHitTwoRules() throws IOException {
+        wildflyConfigFile = """
+                wildfly:
+                  rules:
+                    - id: "test"
+                      title: "WFLY"
+                      directories: [src]
+                      notify: [user1, user2]
+                    - id: "test2"
+                      title: "WFLY"
+                      notify: [user1, user2]
+                  format:
+                    title:
+                      enabled: false
+                    commit:
+                      enabled: false
+                """;
+        gitHubJson = GitHubJson.builder(TestConstants.VALID_PR_TEMPLATE_JSON)
+                .title(INVALID_TITLE)
+                .build();
+        mockedContext = MockedGHPullRequestProcessor.builder(gitHubJson.id())
+                .prFile("src/main/java/resource/application.properties")
+                .collaborators(Set.of("user1", "user2"));
```

Comment:
I wonder if it wouldn't be better to split PR and repo parts in the builder.

Reply:
I was thinking about creating perhaps 2 consecutive builders or just 2 separate builders all together. Let me know what would you prefer